server/routes/referrals/referralsController.test.ts

```diff
@@ -69,31 +69,63 @@ describe('GET /intervention/:id/refer', () => {
     })
-
-    it('creates a referral on the interventions service and redirects to the referral form', async () => {
-      const interventionId = '98a42c61-c30f-4beb-8062-04033c376e2d'
-      const serviceUserCRN = 'X123456'
-
-      await request(app)
-        .post(`/intervention/${interventionId}/refer`)
-        .send({ 'service-user-crn': serviceUserCRN })
-        .expect(303)
-        .expect('Location', '/referrals/1/form')
-
-      expect(interventionsService.createDraftReferral).toHaveBeenCalledWith('token', serviceUserCRN, interventionId)
+    describe('For a Single Referral', () => {
+      const serviceCategory = serviceCategoryFactory.build({
+        id: 'b33c19d1-7414-4014-b543-e543e59c5b39',
+        name: 'social inclusion',
+      })
+      const intervention = interventionFactory.build({
+        serviceCategories: [serviceCategory],
+      })
+      beforeEach(() => {
+        interventionsService.getIntervention.mockResolvedValue(intervention)
+      })
+      it('creates a referral on the interventions service and redirects to the referral form', async () => {
+        await request(app)
+          .post(`/intervention/${interventionId}/refer`)
+          .send({ 'service-user-crn': serviceUserCRN })
+          .expect(303)
+          .expect('Location', '/referrals/1/form')
+        expect(interventionsService.createDraftReferral).toHaveBeenCalledWith('token', serviceUserCRN, interventionId)
+      })
+      it('should update referral with service user and service Category', async () => {
+        await request(app).post(`/intervention/${interventionId}/refer`).send({ 'service-user-crn': 'X123456' })
+        expect(interventionsService.patchDraftReferral).toHaveBeenCalledWith('token', '1', {
+          serviceUser,
+          serviceCategoryIds: [serviceCategory.id],
+        })
+      })
     })
-
-    it('updates the newly-created referral on the interventions service with the found service user', async () => {
-      await request(app)
-        .post('/intervention/98a42c61-c30f-4beb-8062-04033c376e2d/refer')
-        .send({ 'service-user-crn': 'X123456' })
-
-      expect(interventionsService.patchDraftReferral).toHaveBeenCalledWith('token', '1', {
-        serviceUser,
+    describe('For a Cohort Referral', () => {
+      const serviceCategory1 = serviceCategoryFactory.build({
+        id: 'cb0a3633-4367-49ad-94a7-2274839b216e',
+        name: 'accommodation',
+      })
+      const serviceCategory2 = serviceCategoryFactory.build({
+        id: 'b33c19d1-7414-4014-b543-e543e59c5b39',
+        name: 'social inclusion',
+      })
+      const intervention = interventionFactory.build({
+        serviceCategories: [serviceCategory1, serviceCategory2],
+      })
+      beforeEach(() => {
+        interventionsService.getIntervention.mockResolvedValue(intervention)
+      })
+      it('should update referral with service user and redirect to ...', async () => {
```

Comment:
Sorry if I didn't explain this quite right earlier, this should redirect to the referral form like we do above - the form will be a bit different to the one we use for a single-service referral - the page where we select the relevant service categories comes as step 2 in the task list, if I remember correctly, but we can add that in later.

Reply:
Ok. I misunderstood it as redirecting to a new page.
I've changed it as you suggested